crates/core/src/addon.rs

```diff
@@ -344,6 +365,13 @@ impl Addon {
                     file_id: Some(file.id),
                 };
 
+                let file_folders: Vec<AddonFolder> = addon_folders
+                    .iter()
+                    .filter(|f| file.modules.iter().any(|m| m.foldername == f.id))
+                    .cloned()
+                    .collect();
+                folders.extend(file_folders);
+
```

Comment:
I run this on every file, instead of only the "main" file.
This way we ensure we get all the dependencies. Late `dedup` it.